### DISCUSSION TITLE
How to intercept window closing events？

### DISCUSSION THREAD

**Discussion by dibadibadiudiu** [tauri-apps/tauri#1856] [Q&A]
Thank you for your attention,I hope hide the window when users click close,This is my code. It can't intercept the close event.🙏

OS：mac m1 bigsur

```
.on_window_event(|event| match event.event() {
      WindowEvent::CloseRequested => {
        event.window().hide().unwrap();
      }
      _ => {} // event.window().hide().unwrap();
    })
```

**Comment by amrbashir** [comment]
You can hide the window decorations and make a custom titlebar, this way you have total control over whether your close button should hide or close the window.
